sql/core/src/test/scala/org/apache/spark/sql/DataFrameAsSchemaSuite.scala

```diff
@@ -46,15 +46,11 @@ class DataFrameAsSchemaSuite extends QueryTest with SharedSparkSession {
     checkAnswer(df, Row("b"))
   }
 
-  test("negative: column not found") {
```

Comment:
can we keep the old test but change the column nullability to false?

Reply:
Ah I see, you add a new test for both columns and fields. I think it's better to keep the old tests because this test suite put column tests one after each other, then field tests.

Reply:
@cloud-fan - sure, updated.